### DISCUSSION TITLE
MWE using Gaussian blur filter

### DISCUSSION THREAD

**Discussion by enricofacca** [cduck/drawSvg#96] [Q&A] [answered]
Can someone provide a Minimal Working Example where a Gaussian filter is applied to blur the edges of a shape?
Like the filter described [here](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feGaussianBlur).

I tried using the Filter and FilterItem classes but I did not succeed.

Thank you in advance.

Enrico

**Comment by cduck** [comment, marked as answer]
This is a good question.  Filters can be a bit tricky.  Here is [that example](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/feGaussianBlur) translated to drawsvg:

```python
d = draw.Drawing(230, 120)

# Circle
d.append(draw.Circle(60, 60, 50, fill='green'))

# Blurred circle
blur = draw.Filter()
blur.append(draw.FilterItem('feGaussianBlur', in_='SourceGraphic', stdDeviation=5))
d.append(draw.Circle(170, 60, 50, fill='green', filter=blur))

d.save_svg('out.svg')
d
```
![Output SVG](https://user-images.githubusercontent.com/2476062/232523674-dae62c5d-5b9a-43ab-8ee4-facaf408334d.svg)

SVG file content (from `print(d.as_svg())`):
```svg
<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
     width="230" height="120" viewBox="0 0 230 120">
<defs>
<filter id="d0">
<feGaussianBlur in="SourceGraphic" stdDeviation="5"></feGaussianBlur>
</filter>
</defs>
<circle cx="60" cy="60" r="50" fill="green" />
<circle cx="170" cy="60" r="50" fill="green" filter="url(#d0)" />
</svg>
```

Notes:
- You need to use the keyword argument `in_` instead of `in` because `in` is a reserved keyword in Python.  Adding an underscore to the end also works when you need to use the SVG `class=""` or other attriributes.
- `d.raster()`, `d.save_mp3()`, etc. will ignore any filter effects because the CairoSVG library does not support them.  Only `.svg` file output will show the blur effect.
- When you ask questions like this in the future, it is helpful if you include a short example of the code you tried and its output.

**Reply by enricofacca** [reply]
Thank you very much for the nice and fast response.
You also answered to my second question on why the filter was not applied in the svg to png conversion.
